src/services/socketIoServices.ts

```diff
@@ -0,0 +1,169 @@
+'use client'
+
+import { io } from 'socket.io-client'
+import { useEffect, useState } from 'react'
+
+export type ListenerRes = {
+	messageServer: string
+	roomIdServer: string
+	userNameServer: string
+}
+
+export interface Event extends ListenerRes {
+	timeStamp: string
+}
+
+const socket = io()
+
+export function socketRoomEmitter(
+	emitterName: string,
+	message: string,
+	roomId: string,
+	userName: string,
+) {
+	socket.emit(emitterName, message, roomId, userName)
+}
+
+export function useSocketListener(listenerName: string) {
```

Comment:
q: Where is this used? I don't see it being used anywhere.

Reply:
It's being used in the room pages. I just haven't committed those changes.